rebac-admin-backend/v1/identity_providers.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// GetAvailableIdentityProviders returns the list of supported identity providers.
+// (GET /authentication/providers)
+func (h handler) GetAvailableIdentityProviders(w http.ResponseWriter, req *http.Request, params resources.GetAvailableIdentityProvidersParams) {
+	ctx := req.Context()
+
+	identityProviders, err := h.IdentityProviders.ListAvailableIdentityProviders(ctx, &params)
+	if err != nil {
+		response := h.IdentityProvidersErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := resources.GetAvailableIdentityProvidersResponse{
+		Data:   identityProviders.Data,
+		Status: http.StatusOK,
+	}
+
+	writeResponse(w, http.StatusOK, response)
+
+}
+
+// GetIdentityProviders returns a list of known authentication providers.
```

Comment:
Here we say "known", but on the interface it was "registered". Can we please use one of these to avoid confusion?